mirror_builder/wheels.py

```diff
@@ -1,36 +1,83 @@
 import logging
+import platform
+import venv
 
 from . import external_commands, overrides, server
 
 logger = logging.getLogger(__name__)
 
 
-def build_wheel(ctx, req_type, req, resolved_name, why, sdist_root_dir):
+def build_wheel(ctx, req_type, req, resolved_name, why, sdist_root_dir, build_dependencies):
     logger.info('building wheel for %s', resolved_name)
     builder = overrides.find_override_method(req.name, 'build_wheel')
     if not builder:
         builder = _default_build_wheel
-    wheel_filenames = builder(ctx, req_type, req, resolved_name, why, sdist_root_dir)
+    wheel_filenames = builder(ctx, req_type, req, resolved_name, why, sdist_root_dir,
+                              build_dependencies)
     for wheel in wheel_filenames:
         server.add_wheel_to_mirror(ctx, sdist_root_dir.name, wheel)
     ctx.add_to_build_order(req_type, req, resolved_name, why)
-    logger.info('built wheel for %s', resolved_name)
+    if wheel_filenames:
+        logger.info('built wheel for %s', resolved_name)
+    return
 
 
-def _default_build_wheel(ctx, req_type, req, resolved_name, why, sdist_root_dir):
+def _default_build_wheel(ctx, req_type, req, resolved_name, why, sdist_root_dir,
+                         build_dependencies):
+    build_env = BuildEnvironment(ctx, sdist_root_dir.parent, build_dependencies)
     cmd = [
-        'firejail',
-        '--net=none',
-        '--join=local-wheel-server',
-        'pip', '-vvv',
+        # FIXME: Need to cope with build dependencies on rust packages before
+        #        we can fully isolate the wheel build step.
+        # 'firejail',
+        # '--net=none',
```

Comment:
If we're to put in a temporary hack to allow some packages network access, I'd prefer to restrict that to the packages that need it